pkg/chartutil/expand.go

```diff
@@ -81,3 +81,67 @@ func ExpandFile(dest, src string) error {
 	defer h.Close()
 	return Expand(dest, h)
 }
+
+// Expand uncompresses and extracts a chart into the specified directory.
+// Return the directory when extracting successfully.
+func ExpandWithPath(dir string, r io.Reader) (string, error) {
+	gr, err := gzip.NewReader(r)
```

Comment:
there is a significant amount of duplicate code shared with `Expand` save for `expandDir = fullDir`. I understand that we don't want to break the public signatures, but is there any way we can refactor the two functions so we have less code duplication overall?